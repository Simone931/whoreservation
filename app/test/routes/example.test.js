const example = require('../../routes/example');
const fastify = require('fastify');
const elkLogService = example.__get__('elkLogService');

describe('example is loaded', () => {
  let fastifyInstance = null;
  beforeEach(async () => {
    fastifyInstance = fastify();
    fastifyInstance.sendLogMessage = jest.fn();
  });

  afterAll(() => {
    fastifyInstance.close();
  });

  it('GET returns 200', async () => {
    const spy = jest.spyOn(fastifyInstance, 'sendLogMessage');
    await example(fastifyInstance);
    const response = await fastifyInstance.inject({ method: 'GET', url: '/example' });
    expect(response.statusCode).toEqual(200);
    const payload = JSON.parse(response.payload);
    expect(spy).toHaveBeenCalledWith(elkLogService.elkLogLevel.INFO, 'Send the response');
    expect(payload).toMatchSnapshot({
      code: '178172127981',
      cognome: 'Example',
      nome: 'Hello',
    });
  });

  it('POST returns 404', async () => {
    await example(fastifyInstance);
    const response = await fastifyInstance.inject({ method: 'POST', url: '/' });
    expect(response.statusCode).toEqual(404);
    expect(response.payload).toMatchSnapshot();
  });
});
