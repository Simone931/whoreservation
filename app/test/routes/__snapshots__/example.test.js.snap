// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`example is loaded GET returns 200 1`] = `
Object {
  "code": "178172127981",
  "cognome": "Example",
  "nome": "Hello",
}
`;

exports[`example is loaded POST returns 404 1`] = `"{\\"message\\":\\"Route POST:/ not found\\",\\"error\\":\\"Not Found\\",\\"statusCode\\":404}"`;
